spec/hmac_auth_spec.rb

```diff
@@ -37,33 +52,110 @@
     let(:path) { '/foo' }
     let(:env) do
       Faraday::Env.new.tap do |c|
+        c.method = :get
         c.request_headers = {}
         c.url = URI(path)
       end
     end
     let(:request_headers) { middleware.call(env).request_headers }
     let(:secrets) { { 'key' => SecureRandom.hex } }
-    let(:rack_env) do
-      {
-        'HTTP_X_REQUEST_ID' => request_headers['X-Request-ID'],
-        'HTTP_X_SIGNATURE' => request_headers['X-Signature'],
-        'PATH_INFO' => path
-      }
-    end
 
     specify 'header encoding' do
       expect(request_headers['X-Request-ID']).to start_with('key-')
       expect(request_headers.key?('X-Signature')).to eq(true)
     end
 
+    let(:stubs) do
+      Faraday::Adapter::Test::Stubs.new do |stub|
+        stub.get('/endpoint', &stub_verifier)
+      end
+    end
+
+    let(:stub_verifier) do
+      lambda do |env|
+        [200, {}, '{}']
+      end
+    end
+
+    let(:connection_options) do
+      { secret: 'secret', key: 'key', adapter: [:test, stubs] }
+    end
+
+    let(:connection) { HMACAuth::Faraday.connection(connection_options) }
+
+    specify 'get request' do
+      connection.get('/endpoint').tap do |response|
```

Comment:
took these tests in a different direction (just amended the commit to remove this).